web/modules/custom/farm_nfa/src/Form/ForestPlanGfwForm.php

```diff
@@ -27,9 +33,17 @@ public function buildForm(array $form, FormStateInterface $form_state) {
     // Set the form title.
     $form['#title'] = $this->t('GFW');
 
-    $form['placeholder'] = [
-      '#type' => 'markup',
-      '#markup' => '(GFW form placeholder)',
+    $form['gfw_map'] = [
+      '#type' => 'farm_map',
+      '#map_type' => 'farm_nfa_plan_locations',
+      '#map_settings' => [
+        'plan' => $this->routeMatch->getRawParameter('plan')
```

Comment:
You're quite right here, I was giving you examples of blocks which are not accurate, apologies.

You should be getting the plan by using getParameter, that should be of PlanInterface, you can check if it is before rendering anything and avoid checking regex in Javascript.

You can also pass drupalSettings I think to the library, see https://www.drupal.org/docs/drupal-apis/javascript-api/javascript-api-overview

Reply:
no worries @pcambra , the regex check in Javascript has been already been removed as i am attacahing the library separately now in the form, so we no longer need to do any checks in javascript. though as suggested after using `getParameter` instead of `getRawParameter` i am getting an empty object `{}` for `plan` parameter  instead of the expected plan id that i was getting on using the `getRawParameter`

Reply:
We can live with getRawParameter for the time being but probably need to look at why getParameter is not working.

You're still using:

```
  let planId = window.location.href.split('/')[4];
  let cfrPlanUrl = `${window.origin}/nfa-assets/geojson/${planId}`;
```

When you can pass the plan id along the URL as drupalSetting instead.

Reply:
@pcambra when you said to pass the plan id along the URL as drupalSetting instead, do you mean to pass only the planId
as a drupal setting and then use it in the cfrPlanUrl ? or pass both the planId and cfrPlanUrl as a drupal setting instead ?